packages/lib/src/core/core.ts

```diff
@@ -212,11 +212,17 @@ class Core implements ICore {
      * @internal
      * Create or update the config object passed when AdyenCheckout is initialised (environment, clientKey, etc...)
      */
-    private setOptions = (options: CoreConfiguration): void => {
+    private setOptions = (options: CoreConfiguration, isUpdate = false): void => {
         if (hasOwnProperty(options, 'paymentMethodsConfiguration')) {
             console.warn('WARNING:  "paymentMethodsConfiguration" is supported only by Drop-in.');
         }
 
+        if (!isUpdate && !options.session) {
```

Comment:
I think putting the logic separately in the `update` and `initialize` would be better. Because it keeps the `setOptions`'s single responsibility.

Reply:
But then there's actually _three_ places the logic has to be - which seems like unnecessary duplication to me

Reply:
I think duplication would be okay in some scenarios. I feel adding `isUpdate` flag and checking upon `option.session` adds more complexity to a very simple setter function😄. 
As a result, it does more than just setting the option, it needs to know if it's an update operation and it needs to know if it's a session flow.

Reply:
done